### PR TITLE
Make test module names consistent.

### PR DESCRIPTION
**Pull request by thealmarty**
Now all test modules are named `Tests.hs`. Also fixed some line lengths.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Changelog fragments have been written (if appropriate)
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting master unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-benchmark/lists/test/Sum/Spec.hs [hunk 1]
```diff
@@ -10,7 +10,7 @@ import PlutusBenchmark.Common (Term, cekResultMatchesHaskellValue)
 import PlutusBenchmark.Lists.Sum.Compiled qualified as Compiled
 import PlutusBenchmark.Lists.Sum.HandWritten qualified as HandWritten
 
-import PlutusTx.Test qualified as Tx
+import PlutusTx.Tests qualified as Tx
```
Comment (michaelpj) [inline plutus-benchmark/lists/test/Sum/Spec.hs:13]: This is wrong: `PlutusTx.Test` is a _testlib_ not a file containing tests.

File: plutus-benchmark/marlowe/test/Spec.hs [hunk 2]
```diff
@@ -12,10 +12,10 @@ import PlutusBenchmark.Marlowe.Scripts.RolePayout (rolePayoutValidator)
 import PlutusBenchmark.Marlowe.Scripts.Semantics (marloweValidator)
 import PlutusBenchmark.Marlowe.Types qualified as M
 import PlutusCore.Default (DefaultFun, DefaultUni)
-import PlutusCore.Test (goldenSize, goldenUEvalBudget)
+import PlutusCore.Tests (goldenSize, goldenUEvalBudget)
```
Comment (michaelpj) [inline plutus-benchmark/marlowe/test/Spec.hs:15]: ditto

File: plutus-core/plutus-ir/test/PlutusIR/Analysis/RetainedSize/Tests.hs [hunk 3]
```diff
@@ -9,7 +9,7 @@ import PlutusCore.Quote
 import PlutusCore.Rename qualified as PLC
 import PlutusIR.Analysis.RetainedSize qualified as RetainedSize
 import PlutusIR.Parser
-import PlutusIR.Test
+import PlutusIR.Tests
```
Comment (michaelpj) [inline plutus-core/plutus-ir/test/PlutusIR/Analysis/RetainedSize/Tests.hs:12]: ditto

**Review by michaelpj — CHANGES_REQUESTED** [review]
The testlibs aren't supposed to be called `Tests`, they don't have tests in them!

File: plutus-benchmark/lists/test/Sum/Spec.hs (hunk 1 above)
Comment (thealmarty) [reply plutus-benchmark/lists/test/Sum/Spec.hs:13]: Right! They're more like `Utils`? `plutus-core/testlib/PlutusCore/Generators/Hedgehog` already has `Utils` and frankly it's the same stuff as its `Test` module so may I suggest calling them all `Utils`? And I'll combine the `Hedgehog/Utils` and `Test` module.

Comment (michaelpj) [reply plutus-benchmark/lists/test/Sum/Spec.hs:13]: I think `X.Test` is a reasonable module prefix for a testlib? We could have `X.Test.Utils`, but often we use the parent module to clump stuff together or re-export things. I don't mind that much if you want them to be `X.Test.Utils` though.

Comment (thealmarty) [reply plutus-benchmark/lists/test/Sum/Spec.hs:13]: I'll just undo the changes then.

Comment (michaelpj) [reply plutus-benchmark/lists/test/Sum/Spec.hs:13]: You missed this one?

**Review by michaelpj — APPROVED** [review]

**Comment by thealmarty** [comment]
Anyone has any idea why some tests are failing on the CI? This PR does not change any tests!

**Comment by thealmarty** [comment]
Verified that the tests pass locally for me. Could someone with permission please rerun these? I think it's just a fluke.

**Pull request merged**